d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadLoadBalancer.java

```diff
@@ -291,5 +312,6 @@ public void shutdown(PropertyEventThread.PropertyEventShutdownCallback callback)
     });
 
     _oldLb.shutdown(callback);
+    _loadBalancerThreadPool.shutdown();
```

Comment:
Technically you're getting this as a constructor parameter, so it's not your responsibility to shut it down. You would only have to shut it down if your had initialized it yourself

Reply:
Make senses. Remove the shutdown logic